lib/rspec/hive/query_builder.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'query_builder/row_transformer'
+require_relative 'query_builder/null_strategy'
+require_relative 'query_builder/value_by_type_strategy'
+
+module RSpec
+  module Hive
+    class QueryBuilder
+      def initialize(schema, connection)
+        @schema = schema
+        @connection = connection
+        @partition_hash = {}
+        @rows = []
+        @stubbing = false
+      end
+
+      def partition(hash)
+        spawn.merge_partition!(hash)
```

Comment:
@mknapik I liked your your builder idea, and I left I should bring after all! I took a look on how this stuff is actually implemented on `ActiveRecord` and made the following changes. See what you think. As reference I used these links:
[spawn](https://github.com/rails/rails/blob/52ce6ece8c8f74064bb64e0a0b1ddd83092718e1/activerecord/lib/active_record/relation/spawn_methods.rb)
[method, method!](https://github.com/rails/rails/blob/ad078767fb4d4a78224efaaf541e9bc847cde245/activerecord/lib/active_record/relation/query_methods.rb#L568)

These are included on [`ActiveRecord::Base`](https://github.com/rails/rails/blob/b1879124a82b34168412ac699cf6f654e005c4d6/activerecord/lib/active_record/relation.rb#L17)

Really interesting stuff!